runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5318,6 +5460,160 @@ void chpl_comm_get(void* addr, c_nodeid_t locale, void* raddr,
 }
 
 
+// Max number of threads that can do buffered gets
+#define MAX_BUFF_GET_THREADS 1024
```

Comment:
I still feel like this could be done a lot more cleanly.  It seems like it would be straightforward to define a struct type that contained all the information for a specific thread, and then have each thread add its struct to a chain of all of them, the first time it buffers up a GET.  Then the only global data we'd need would be a head pointer for the chain and a lock to protect it.

Reply:
Think you convinced me. You ok with me opening an issue for that and doing it in a followup PR?

Reply:
Yeah, think that would be okay.